django/thunderstore/repository/models/package_version.py

```diff
@@ -25,13 +25,18 @@ def get_version_png_filepath(instance, filename):
     return f"repository/icons/{instance}.png"
 
 
-class PackageVersionManager(models.Manager):
-    def active(self) -> "QuerySet[PackageVersion]":  # TODO: Generic type
+class PackageVersionQyerySet(QuerySet):
+    def active(self):
         return self.exclude(is_active=False)
 
+    def listed_in(self, community_identifier: str):
+        return self.exclude(
+            ~Q(package__community_listings__community__identifier=community_identifier),
+        )
```

Comment:
I don't know how postgres actually evaluates this, but the way this query is written traverses the relation the wrong way around, and could easily be much more inefficient than it should be. Specifically I think this query will lead to unnecessary table joins, when really we only care about a set of IDs to filter against.

The "correct" solution would be to query the package listings that exist in the community and traverse the relation from that side. I'm fairly certain you can do this with a subquery in case you want to make it queryable through the `PackageVersion` manager.

I don't consider this a blocker per se, but in terms of query architecture it's less than ideal. This somewhat extends to the API view that brought this PackageVersion-centric approach on in the first place, since it could AFAIK query the PackageListing model rather than the PackageVersion model given that it's the actual thing we're interested in "finding".